exonum/src/node/consensus.rs

```diff
@@ -550,11 +565,18 @@ impl NodeHandler {
             .expect("Unable to save transaction to persistent pool.");
 
         let full_proposes = self.state.check_incomplete_proposes(hash);
-        // Go to handle full propose if we get last transaction
+        // Go to handle full propose if we get last transaction.
         for (hash, round) in full_proposes {
-            self.remove_request(&RequestData::Transactions(hash));
+            self.remove_request(&RequestData::ProposeTransactions(hash));
             self.handle_full_propose(hash, round);
         }
+
+        let full_block = self.state.remove_unknown_transaction(hash);
+        // Go to handle full block if we get last transaction
+        if full_block.is_some() {
```

Comment:
`if let` could be better here; it allows to eliminate `full_block.unwrap()` later.